### PR TITLE
Added Tabular Regression Model using Boston Housing dataset

### PR DESCRIPTION
**Pull request by JacopoMangiavacchi**
Following the path of other models in this repo, such as the CIFAR-10, I've added here a Tabular sample project to train a regression model on the Boston Housing dataset.

### CODE REVIEW

**Comment by 8bitmp3** [comment]
@JacopoMangiavacchi This dataset and the example are awesome! Hope you don't mind I amended some bits to make the README in line with the rest of the Examples in /swift-models/. The dataset description is from keras.io/datasets/. The PR in your forked repo is here: https://github.com/JacopoMangiavacchi/swift-models/pull/1 - feel free to edit it. Your contributions and presentations are awesome 🙌

**Comment by 8bitmp3** [comment]
@JacopoMangiavacchi May I also suggest to perhaps rename `Tabular` to `Regression-BostonHousing` or `Regression-BostonHousing-Tabular`. This should aid the discovery process for current and future s4tf users and would look more descriptive under `/Examples`, similar to: `Custom-CIFAR10`, `LeNet-MNIST`, `MobileNet-Imagenette`, `ResNet-CIFAR10`, and `VGG-Imagewoof`. (And if you choose to rename it, some bits in `package.swift` and other places may need to be revised too.)

File: Tabular/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Tabular
```
Comment (8bitmp3) [inline Tabular/README.md]: If it's quicker for you to edit here and ignore my small PR, these are the changes I proposed:

1) More descriptive title - like the other examples:
```
# Regression with tabular Boston housing price dataset
```

File: Tabular/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+This directory contains a regression model trains a Boston house prices dataset.
```
Comment (8bitmp3) [inline Tabular/README.md]: 2) Rewording of the description 1 / x:
```
This example demonstrates how to train a regression model against the [Boston 
housing price dataset](https://archive.ics.uci.edu/ml/machine-learning-databases/housing/).
```

File: Tabular/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+This is a very small dataset with only 506 data points and 13 numeric/categorical features.  Median Value (last column) is usually the target.
```
Comment (8bitmp3) [inline Tabular/README.md]: 3) Rewording of the description 2 / x:
```
There are 506 data points. Samples contain 13 attributes of houses at different 
locations around the Boston suburbs in the late 1970s. Targets are the median 
values of the houses at a location (in k$).
```

(source of the 2nd and 3rd sentences: keras.io/datasets/)

File: Tabular/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+:Creator: Harrison, D. and Rubinfeld, D.L.
```
Comment (8bitmp3) [inline Tabular/README.md]: 4) The link is in the first sentence - remove this bit in lines 7 and 8 (in line with other README.md e.g. ResNet-CIFAR10).

File: Tabular/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+This is a copy of UCI ML housing dataset. https://archive.ics.uci.edu/ml/machine-learning-databases/housing/
+
+## Setup
+
+You'll need [the latest version][INSTALL] of Swift for TensorFlow
```
Comment (8bitmp3) [inline Tabular/README.md]: 5. Another suggestion - update this line - word for word from other READMEs (lines 12-13):

```
To begin, you'll need the [latest version of Swift for 
TensorFlow](https://github.com/tensorflow/swift/blob/master/Installation.md)
```

File: Tabular/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+installed and added to your path. Additionally, the data loader requires Python
+3.x (rather than Python 2.7), `wget`, and `numpy`.
```
Comment (8bitmp3) [inline Tabular/README.md]: 6. I think saying `3.x` is enough:

```
3.x, `wget`, and `numpy`.
```

File: Tabular/README.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+> Note: For macOS, you need to set up the `PYTHON_LIBRARY` to help the Swift for
+> TensorFlow find the `libpython3.<minor-version>.dylib` file, e.g., in
+> `homebrew`.
+
+To train the model on the Boston house prices dataset, run:
```
Comment (8bitmp3) [inline Tabular/README.md]: 7. In line with the rest of READMEs:

```
To train the model, run:
```

File: Tabular/README.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+```
```
Comment (8bitmp3) [inline Tabular/README.md]: 8. I think the other READMEs highlight shell/bash code here, so I'd add `sh`:

```
```sh
```

File: Tabular/README.md (hunk 1 above)
Comment (JacopoMangiavacchi) [reply Tabular/README.md]: Thanks @8bitmp3 I've merged your PR

**Comment by JacopoMangiavacchi** [comment]
> @JacopoMangiavacchi May I also suggest to perhaps rename `Tabular` to `Regression-BostonHousing` or `Regression-BostonHousing-Tabular`...

Absolutely, I'm renaming this with Regression-BostonHousing

File: Examples/Regression-BostonHousing/README.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+There are 506 data points. Samples contain 13 attributes of houses at different 
+locations around the Boston suburbs in the late 1970s. Targets are the median 
+values of the houses at a location (in k$).
+
+## Setup
+
+To begin, you'll need the [latest version of Swift for
+TensorFlow](https://github.com/tensorflow/swift/blob/master/Installation.md)
+installed and added to your path. Additionally, the data loader requires Python
```
Comment (BradLarson) [inline Examples/Regression-BostonHousing/README.md]: You state that the data loader requires Python and wget, but I no longer see those dependencies here. Is this documentation from a previous version or model?

File: Datasets/BostonHousing/BostonHousing.swift [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+    public let numTrainRecords: Int
+    public let numTestRecords: Int
+    public let xTrain: Tensor<Float>
+    public let yTrain: Tensor<Float>
+    public let xTest: Tensor<Float>
+    public let yTest: Tensor<Float>
+
+    static func downloadBostonHousingIfNotPresent() -> String {
+        let remoteURL = URL(string: "https://archive.ics.uci.edu/ml/machine-learning-databases/housing/")!
+        let localURL = URL(fileURLWithPath: "./regression-bostonHousing-batches-bin")
```
Comment (BradLarson) [inline Datasets/BostonHousing/BostonHousing.swift]: We've had problems (especially internally in our build systems) with files being downloaded to a path relative to the executable. Would it be possible to switch this for a location in the local temporary directory, like is specified here: https://github.com/tensorflow/swift-models/blob/master/Datasets/MNIST/MNIST.swift#L35 ?

File: Examples/Regression-BostonHousing/main.swift [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+    
+    @differentiable
+    func callAsFunction(_ input: Tensor<Float>) -> Tensor<Float> {
+        return input.sequenced(through: layer1, layer2, layer3)
+    }
+}
+
+var model = RegressionModel()
+
+//Train Model
```
Comment (BradLarson) [inline Examples/Regression-BostonHousing/main.swift]: Minor nitpick: the "Train Model" and "Evaluate Model" comments don't have a single space before the first word, like is done for all other comments here.

**Pull request merged**